lib/Partitioner/PartitionerUtils.cpp

```diff
@@ -493,6 +493,31 @@ GraphMemInfo getGraphMemInfo(const NodesSet &nodes) {
   return ret;
 }
 
+GraphMemInfo getFunctionMemory(Function *func) {
+  GraphMemInfo graphMem;
+
+  for (auto cons : func->findConstants()) {
+    graphMem.constMemSize += cons->getType()->getSizeInBytes();
+  }
+
+  // Walk thru all the Placeholders in the function to accumulate input and
+  // output mem size.
+  for (auto &place : func->findPlaceholders()) {
+    graphMem.inMemSize += place->getType()->getSizeInBytes();
+  }
+
+  for (auto &node : func->getNodes()) {
+    // If a node is SaveNode then accumulate size for output memory.
+    if (auto *SN = llvm::dyn_cast<SaveNode>(&node)) {
+      graphMem.outMemSize += SN->getOutput().getType()->getSizeInBytes();
+    }
+  }
+  // Output memory got double counted as inputs during Placeholder list scan
+  // so subtract from the input memory size.
+  graphMem.inMemSize -= graphMem.outMemSize;
```

Comment:
This assumes a PH can never be an input and output if a PH is both then we would have the correct output size but input size would be smaller than it should be.
I think with training it's not safe to assume input/output is exclusive since a trainable weight is both an input and output. @SplitInfinity is that correct?

Reply:
A trainable `Placeholder` is considered both an input and an output, yes.

Reply:
Good point. I didn't realize this at all. I modified the code just now to use pre-existing functions to determine Input, Output by determining users for the PH.